Vivek/Robot/Robot.java

```diff
@@ -0,0 +1,135 @@
+
+        this.grid = new Cell[this.rowLength][this.columnLength];
+        populateGrid(gridArray);
+    }
+
+    private void validGridArray(int[][] gridArray) {
+        if (gridArray == null || gridArray.length < 1 || gridArray[0].length < 1) {
+            throw new RuntimeException("Null Grid Array or Invalid Grid Array Length");
+        }
+
+        if (gridArray[0][0] == 1) {
+            throw new RuntimeException("Robot Start value cannot be a Obstacle");
+        }
+
+        if (gridArray[gridArray.length - 1][gridArray[0].length - 1] == 1) {
+            throw new RuntimeException("Robot End value cannot be a Obstacle");
+        }
+    }
+
+    private void populateGrid(int[][] gridArray) {
+        for (int i = 0; i < this.rowLength; i++) {
+            for (int j = 0; j < this.columnLength; j++) {
+                grid[i][j] = Cell.valueToCell(gridArray[i][j]);
+            }
+        }
+    }
+
+    public boolean isCellObstacle(int rowIndex, int colIndex) {
+        validateIndices(rowIndex, colIndex);
+        return grid[rowIndex][colIndex] == Cell.OBSTACLE;
+    }
+
+    public boolean isCellEmpty(int rowIndex, int colIndex) {
+        validateIndices(rowIndex, colIndex);
+        return grid[rowIndex][colIndex] == Cell.EMPTY;
+    }
+
+    private void validateIndices(int rowIndex, int colIndex) {
+        if (rowIndex < 0 || rowIndex >= rowLength) {
+            throw new RuntimeException(String.format("Invalid Row Index: %s. RowLength: %s", rowIndex, rowLength));
+        }
+
+        if (colIndex < 0 || colIndex >= columnLength) {
+            throw new RuntimeException(String.format("Invalid Column Index: %s. ColumnLength: %s", colIndex, columnLength));
+        }
+    }
+
+    public int getRowLength() {
+        return rowLength;
+    }
+
+    public void setRowLength(int rowLength) {
+        this.rowLength = rowLength;
+    }
+
+    public int getColumnLength() {
+        return columnLength;
+    }
+
+    public void setColumnLength(int columnLength) {
+        this.columnLength = columnLength;
+    }
+
+
+    private enum Cell {
+        OBSTACLE(1),
+        EMPTY(0);
+
+        int value;
+
+        Cell(int value) {
+            this.value = value;
+        }
+
+        public static Cell valueToCell(int value) {
+            for (Cell cell : Cell.values()) {
+                if (value == cell.value) {
+                    return cell;
+                }
+            }
+
+            throw new RuntimeException("Invalid Cell Value: " + value);
+        }
+    }
+}
+
+class UniquePath {
+    public int findUniquePaths(int[][] gridArray) {
+        Grid grid = new Grid(gridArray);
+        int[][] uniquePathMatrix = generateUniquePathMatrix(grid);
+        return uniquePathMatrix[grid.getRowLength()][grid.getColumnLength()];
+    }
+
+    private int[][] generateUniquePathMatrix(Grid grid) {
+        int[][] uniquePathMatrix = new int[grid.getRowLength() + 1][grid.getColumnLength() + 1];
```

Comment:
You use a lot of `+1` and `-1` which might make it easy to have an off by one error in an interview. I think it generally won't matter cause talking through it the interviewer will know that you understand the problem.

Reply:
I think I could have written a comment to explain it better. It was a Guard row and column so that I didn't have to deal with a bunch of if...else conditions for edge row and column.